src/library/joiner_session.cpp

```diff
@@ -207,8 +208,17 @@ Error JoinerSession::SendJoinFinResponse(const coap::Request &aJoinFinReq, bool
     joinFin.SetSubType(MessageSubType::kJoinFinResponse);
     SuccessOrExit(error = mCoap.SendResponse(aJoinFinReq, joinFin));
 
+    LOG_INFO(LOG_REGION_JOINER_SESSION, "session(={}) sent JOIN_FIN.rsp: accepted={}", static_cast<void *>(this),
```

Comment:
Could it be a security pitfall to print these messages as logs?

Reply:
What about make this optional with a compile flag?

Reply:
Agree that a compile flag is needed. Just like in OT, we only print these logs when REFERENCE_DEVICE is ON.